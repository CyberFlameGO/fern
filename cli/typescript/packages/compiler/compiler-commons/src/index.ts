export type { FernFile } from "./FernFile";
export type { RelativeFilePath } from "./RelativeFilePath";
export type { CompilerStage, FailedStageResult, StageResult, SuccessfulStageResult } from "./stage";
