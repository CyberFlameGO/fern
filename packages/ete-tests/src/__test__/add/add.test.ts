import { WORKSPACE_DEFINITION_FILENAME } from "@fern-api/compiler-commons";
import execa from "execa";
import { mkdir, readFile, rm } from "fs/promises";
import path from "path";

const GENERATED_DIR = path.join(__dirname, "generated");
const GENERATED_API_DIR = path.join(GENERATED_DIR, "api");

describe("fern add tests", () => {
    it("fern add", async () => {
        await rm(GENERATED_DIR, { force: true, recursive: true });
        await mkdir(GENERATED_DIR);
        await init();
        await add("java");
        await add("typescript");
        await add("postman");
        const fileContents = await readFile(path.join(GENERATED_API_DIR, WORKSPACE_DEFINITION_FILENAME));
        expect(fileContents.toString()).toMatchSnapshot();
    }, 10_000);
});

async function init() {
    const init = execa("node", [path.join(process.cwd(), "../cli/cli"), "init"], {
        env: {
            NODE_ENV: "development",
        },
        cwd: GENERATED_DIR,
    });
    init.stdout?.pipe(process.stdout);
    init.stderr?.pipe(process.stderr);
    await init;
}

async function add(generator: string) {
    await execa("node", [path.join(process.cwd(), "../cli/cli"), "add", generator], {
        env: {
            NODE_ENV: "development",
        },
        cwd: GENERATED_DIR,
    });
}
