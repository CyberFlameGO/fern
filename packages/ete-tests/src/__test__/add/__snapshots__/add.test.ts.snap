// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fern add tests fern add 1`] = `
"name: Blog Post API
definition: src
plugins:
  - name: fernapi/fern-java
    version: 0.0.30
    output: generated-java
    config:
      packagePrefix: com
      mode: client_and_server
  - name: fernapi/fern-typescript
    version: 0.0.71
    output: generated-typescript
    config:
      mode: client
  - name: fernapi/fern-postman
    version: 0.0.3
    output: generated-postman
"
`;
